### ISSUE TITLE
Add tax identification / number field

### ISSUE THREAD

**Issue by SukcesStrony** [woocommerce/woocommerce#14997]
Almost in every country when purchasing you need to supply some kind of tax ID - especially if you're buying as a company.

**I ask that you add a tax identification field (type: text) to the user profiles and checkout form, with a filter which would let people implement verification if needed and change the naming of the field if needed (while still being available globally for the translators).**

Currently there are plugins which add that field but only for European VAT purposes, while other countries may require other numbers. These plugins are bad however: there's no way to edit the tax ID field by the user or admin, it gets included outside of the main personal details table in the e-mails etc... It's really bad. The "Additional info" field doesn't really cut it - people forget to put that in, it's not editable, not saved in the user profile etc...

Please let the plugin translators decide on how to name this field. European countries would name it something like "VAT-UE number or TAX ID", other countries would use their tax ID names. Please enable this field also even if the company name field is disabled (some countries might require that).

As this shouldn't break anything, this could come in earlier than WC 4.0.

**I think this has a very high priority for almost everyone in the whole world.**

**Comment by mikejolley** [comment]
Please log your vote on http://ideas.woocommerce.com/forums/133476-woocommerce/suggestions/6794538-company-id-vat-id-to-be-stored-in-customer-accou I don't feel it's a high enough priority for us right now.

There are several solutions; the plugins you mention, checkout editors etc. "European countries" have the EU VAT extension which is designed for this.

Not *all* stores need this feature. b2b it may be more important, but not for b2c. And I've never seen field on US stores, only EU. Why force a field on all users?

**Comment by SukcesStrony** [comment]
Even though there aren't many votes for this (honestly it's hard to find it buried within over 1000 other ideas in this category), I find it to be a core element of every e-commerce system and not something that should be done with a plugin, especially in this case where there's simply no plugin that allows editing of this field etc...

I'm not sure about B2C in various countries, but it seems this is a must for all B2B sells, since the invoices need to address a company by its ID. Surely for all of the EU & Australia that's a must, probably in most other countries too. And the EU is the largest market on the planet, so if it was only for it, it still would make sense. But it's obvious B2B requires tax IDs in almost all countries for invoicing.

Adding this field wouldn't _force_ it - it would be optional to fill it out, just like the company's name field. Since we have a company's name field, there should also be a tax ID field, since companies have tax IDs. Right now the company's name alone is not enough to support B2B and in real life almost every store also wants to have the option to sell to businesses.

Please reconsider, preferably with a 2nd opinion.

**Comment by mikejolley** [comment]
Unless you're dealing with EU VAT or only businesses, why would you collect it? We should be optimising checkout and removing fields, not adding more. And there are existing solutions; it's not like this is impossible to add if needed.

**Comment by SukcesStrony** [comment]
If you're doing B2C, you probably don't need any IDs. Making this field optional with a clear label for each country done by translators, like "VAT-UE number for companies only" in Poland or China, would make it clear. It's basically an addition to the "Company name" field, used mostly by companies.

I've done some fast research on the most populated countries in the world and it looks like the USA & Canada don't require tax numbers on invoices, while the EU, China, Australia, Russia, India, Indonesia, Brazil do require tax IDs on invoices. Those countries alone combined account for **49% of the world's population**, not mentioning other countries. The more people in a country, the more companies it has. Therefore, you can safely say that this field is required by over half of the people, or at least by the companies they run.

I see the point in optimizing, but I strongly feel - and the numbers say the same - that this is needed as much as the company name field, right out of the box, not as a faulty plugin.

So, again, please reconsider :)

**Comment by lkraav** [comment]
@mikejolley please see https://quaderno.io/blog/digital-taxes-around-world-know-new-tax-rules/

**tl;dr** for digital products, have to collect VAT/GST taxes from everybody and everything, regardless of location - can remove tax only if valid tax ID is provided.

Correct me if I'm wrong, but to me it looks like WC (still as of 3.5) out of the box doesn't seem ready for recent (few+ years) tax rules for digital product sales, worldwide.
